@types/index.d.ts

```diff
@@ -0,0 +1,185 @@
+     * html: true,
+     * ```
+     */
+    html?: boolean
+
+    /** (mandatory)
+
+     * The callback must be either:
+     * - Atom command: a string or array of  strings,
+     * - a custom callback function,
+     * - or an object where the keys are key modifiers (alt, ctrl or shift) and the values are commands or custom functions
+     *
+     * ## Example:
+     * ```ts
+     * callback: 'application:about',
+     * ```
+     *
+     *
+     * ## Example - Callback with modifiers
+     * ```ts
+     * callback: {
+     *    '': 'application:cmd-1',      // Without modifiers is default action
+     *    'alt': 'application:cmd-2',
+     *    'ctrl': 'application:cmd-3',  // With function callback
+     *    'shift'(data) {
+     *      console.log(data);
+     *    },
+     *    'alt+shift': 'application:cmd-5',       // Multiple modifiers
+     *    'alt+ctrl+shift': 'application:cmd-6'   // All modifiers
+     *  },
+     * data: 'foo'
+     * ```
+     */
+    callback:
+        | string
+        | Array<string>
+        | ((data?: any) => void)
+        | { [modifier: string]: string }
+        | { [modifier: string]: (data?: any) => void }
+
+    /** `data` can be passed as the input argument of callback,  If callback is of type
+     * - `(data: any) => void)` or
+     * - `{ [modifier: string]: ((data: any) => void) }`,
+     *
+     */
+    data?: any
+
+    /** (optional) defaults to `50` */
+    priority?: number
+
+    /** (optional)
+     * The tooltip option may be a string or an object that is passed to Atom's TooltipManager
+     */
+    tooltip?: string | TooltipManager
+
+    /** (optional) Color of the button */
+    color?: string
+
+    /** (optional) Color of the button's background */
+    background?: string
+
+    /** Buttons can be styled with arbitrary CSS through classes.
+     * An example of how the class can be used is show below.
+     *
+     * ## Example:
+     * ```ts
+     * class: 'my-awesome-class'
+     * ```
+     *
+     * ## Example:
+     * ```ts
+     * class: ['multiple', 'classes', 'also', 'works']
+     * ```
+     */
+    class?: string | Array<string>
+}
+
+export declare interface SpacerOptions {
+    /** (optional) defaults to `50` */
+    priority?: number
+}
+
+export declare class ToolBarManager {
+    /** Adds a button. The input to this function is a `ButtonOptions` object */
+    addButton(options: ButtonOptions): void
+
+    /** Adds a spacer. Optionally, you can pass a `SpacerOptions` object */
+    addSpacer(options?: SpacerOptions): void
+
+    /** Use the method removeItems to remove the buttons added by your package. This is particular useful in your package deactivate method, but can be used at any time. */
+    removeItems(): void
+
+    /** The onDidDestroy method takes a function that will be called when the tool-bar package is destroyed. This is useful if your package needs to do cleanup when the tool-bar is deactivated but your package continues running. */
```

Comment:
Maybe wrap this documentation to reduce the line length?

Reply:
Done in https://github.com/suda/tool-bar/pull/289/commits/5da3bf1d82a54254edb2f177b0f0c13d45dab92c